src/flask_cognito_lib/services/cognito_svc.py

```diff
@@ -96,11 +96,16 @@ def exchange_code_for_token(
             "code_verifier": code_verifier,
         }
 
+        if self.cfg.user_pool_client_secret:
```

Comment:
Please could you add a comment above this `if` block to explain why this is here (i.e. Public vs Confidential client)